src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,75 @@
+package nextstep.subway.domain;
+
+import nextstep.subway.applicaion.dto.StationResponse;
+import nextstep.subway.exception.ValidationException;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Embeddable
+public class Sections {
+    @OneToMany(mappedBy = "line",
+            cascade = {CascadeType.PERSIST, CascadeType.MERGE},
+            orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+    private int DELETION_MINIMUM_SIZE = 1;
+
+    public Sections() {
+    }
+
+    public List<Section> getSections() {
+        return sections;
+    }
+
+    public Section getLastSection() {
+        return sections.stream().reduce((first, second) -> second).orElse(null);
+    }
+
+    public void addSection(Section section) {
+        sections.add(section);
+    }
+
+    public List<Station> createStationList() {
+        return this.sections.stream()
+                .flatMap(s -> Stream.of(s.getUpStation(), s.getDownStation()))
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
+    public List<StationResponse> createStationResponseList() {
+        return createStationList().stream()
+                .map(station -> new StationResponse(station.getId(), station.getName()
+                        , station.getCreatedDate(), station.getModifiedDate()))
+                .collect(Collectors.toList());
+    }
+
+    public void deleteSection(Long stationId) {
+        Section lastSection = getLastSection();
+        validateIsExistSection(lastSection);
+        validateLastSection(lastSection, stationId);
+        validateMinimumSizeOfSection();
+        sections.remove(lastSection);
+    }
+
+    private void validateIsExistSection(Section lastSection) {
+        if(lastSection == null) {
+            throw new ValidationException("구간이 존재하지 않습니다.");
+        }
+    }
+
+    private void validateLastSection(Section lastSection, Long stationId) {
+        if(lastSection.getDownStation().getId() != stationId) {
```

Comment:
이건 `Section` 이 판단하도록 하면 어떨까요? 🤔 

하나 더 말씀드리면 primitive type 과 다르게 Long type 은 wrapper class 이기 때문에 nullable 합니다.
따라서 
```suggestion
        if(lastSection.getDownStation().getId().equals(stationId)) {
```
와 같이 정의하시는 더 좋습니다.